popup/popup.js

```diff
@@ -49,28 +56,68 @@ function listenForClicks() {
     });
 }
 
-async function darkModeSwitch(){
-    if(window.darkMode){
-        await browser.tabs.removeCSS({file:"./../dark-mode/dark-mode.css"})
-    }else{
-        await browser.tabs.insertCSS({file:"./../dark-mode/dark-mode.css"})
+function sendMessageToTabs(tabs, data) {
+    /**
+     * Send Message to the content-script
+     */
+    for (let tab of tabs) {
+        browser.tabs.sendMessage(
+            tab.id,
+            data
+        ).then(response => {
+            // alert(JSON.stringify(response));
+            if (response.DarkMode == "Off") {
+                window.darkMode = false;
+            }
+            else {
+                window.darkMode = true;
+            }
+        }).catch(onError);
     }
-    window.darkMode = !window.darkMode
 }
 
+function onError(error) {
+    console.error(`Error: ${error}`);
+}
 
-// to revisit...
+async function darkModeSwitch() {
+    // console.log(window.darkMode);
+    if (window.darkMode) {
+        // await browser.tabs.removeCSS({ file: "./../dark-mode/dark-mode.css" });
+        const tabs = await browser.tabs.query({
+            currentWindow: true,
+            active: true
+        }).catch(onError);
+        sendMessageToTabs(tabs, { "DarkMode": "Off" });
+    } else {
+        // await browser.tabs.insertCSS({ file: "./../dark-mode/dark-mode.css" });
+        const tabs = await browser.tabs.query({
+            currentWindow: true,
+            active: true
+        }).catch(onError);
+        sendMessageToTabs(tabs, { "DarkMode": "On" });
+    }
+    // window.darkMode = !window.darkMode
+}
+
+// to revisit... (I don't think we need this anymore)

```

Comment:
Yeah I think so too, great job of the sending messages to and from background scripts